Cannot build Web UI during install-dev.sh: JavaScript heap out of memory

### Main idea

### Version
`26.5.0-alpha.0` (Current `main` branch, testing via `install-dev.sh`)

### Frquency & Envrionment
1. 100% reproducible on a fresh machine doing a cold build without NODE_OPTIONS exported.
2. OS: Windows 11 (WSL2 Ubuntu 26.04 LTS)
3. Hardware: Mini PC 6850H, 24GB RAM (12GB allocated to WSL2)
4. Note: Testing the full development stack installation via ./scripts/install-dev.sh.

### Reproduction Steps
1. Set up a clean Ubuntu environment (e.g., WSL2 on a Windows host) without any global Node.js configuration.
2. Clone the `backend.ai` repository.
3. Execute `./scripts/install-dev.sh` to initialize the local development environment.
4. The script proceeds to build the frontend packages. When it reaches the `react build` step using `vite`, it crashes due to V8 heap memory constraints.

### Expected Result
The `install-dev.sh` script should completely build the Web UI and start the core infrastructure without manual Node.js heap memory intervention.

### Actual Result
The Web UI build fails and aborts with an exit code 134. Below is the exact terminal log showing the process hitting the ~2GB default V8 memory limit:

```text
react build: vite v6.4.2 building for production...
react build: transforming...
react build: <--- Last few GCs --->
react build: [33414:0x211e6000]    79974 ms: Mark-Compact 2034.6 (2052.7) -> 2030.6 (2050.5) MB, pooled: 0 MB, 864.79 / 0.00 ms  (average mu = 0.120, current mu = 0.024) allocation failure; scavenge might not succeed
react build: [33414:0x211e6000]    80867 ms: Mark-Compact 2033.9 (2051.7) -> 2031.9 (2051.7) MB, pooled: 0 MB, 880.77 / 0.00 ms  (average mu = 0.069, current mu = 0.014) allocation failure; scavenge might not succeed
react build: FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
react build: [ELIFECYCLE] Command failed with exit code 134.
react build: Failed
/home/newparksung/backend.ai/src/ai/backend/webui/react:
[ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL] backend-ai-webui-react@26.5.0-alpha.0 build: `pnpm run build:only && cp -r ./build/* ../build/web/`
```

<img width="1973" height="923" alt="Image" src="https://github.com/user-attachments/assets/1ca65386-b6c7-4079-99d7-4b9c6355cfac" />

### Suggested Workaround
Exporting the NODE_OPTIONS environment variable globally before running the script resolves the issue completely:

```text
export NODE_OPTIONS="--max-old-space-size=8192"
./scripts/install-dev.sh
```


### Alternative ideas

_No response_

### Anything else?

_No response_

--- a/changes/12250.fix.md
+++ b/changes/12250.fix.md
@@ -1,0 +1,1 @@
+Scope a larger Node.js heap size to the editable Web UI build in `install-dev.sh` so a cold install no longer fails with a JavaScript heap out-of-memory error on low-RAM machines.

--- a/scripts/install-dev.sh
+++ b/scripts/install-dev.sh
@@ -778,7 +778,11 @@ install_editable_webui() {
   fi
   pushd ./packages/backend.ai-ui && pnpm install && popd
   pnpm i
-  make compile
+  # The production Web UI build can exceed Node's ~2GB default old-space heap and
+  # crash with "JavaScript heap out of memory" (exit code 134) on low-RAM machines
+  # (e.g. WSL2). Scope a larger heap to the build only, so users don't have to
+  # export NODE_OPTIONS globally before running this script.
+  NODE_OPTIONS="${NODE_OPTIONS:---max-old-space-size=4096}" make compile
   cd ../../../..
 }
 

--- a/scripts/install-dev.sh
+++ b/scripts/install-dev.sh
@@ -781,8 +781,10 @@ install_editable_webui() {
   # The production Web UI build can exceed Node's ~2GB default old-space heap and
   # crash with "JavaScript heap out of memory" (exit code 134) on low-RAM machines
   # (e.g. WSL2). Scope a larger heap to the build only, so users don't have to
-  # export NODE_OPTIONS globally before running this script.
-  NODE_OPTIONS="${NODE_OPTIONS:---max-old-space-size=4096}" make compile
+  # export NODE_OPTIONS globally before running this script. We append rather than
+  # replace, so an explicit user NODE_OPTIONS (e.g. a higher --max-old-space-size)
+  # still wins via Node's last-flag-wins rule.
+  NODE_OPTIONS="${NODE_OPTIONS:-} --max-old-space-size=4096" make compile
   cd ../../../..
 }
 

